geth/jail/jail.go

```diff
@@ -296,8 +297,26 @@ func newJailErrorResponse(err error) string {
 }
 
 // newJailResultResponse returns a string that is a valid JavaScript code.
-// Marshaling is not required as result.String() produces a string
-// that is a valid JavaScript code.
-func newJailResultResponse(result otto.Value) string {
-	return `{"result": ` + result.String() + `}`
+func newJailResultResponse(input otto.Value) string {
```

Comment:
I would love to simplify this. `otto.Value` has a method `Export()` which should return Go-native data type. Maybe we can do it like this:
```go
func newJailResultResponse(result otto.Value) string {
	exportedValue, err := result.Export()
	if err != nil {
		return newJailErrorResponse(err)
	}

	data, err := json.Marshal(struct {
		Result interface{} `json:"result"`
	}{
		Result: exportedValue,
	})
	if err != nil {
		return newJailErrorResponse(err)
	}

	return string(data)
}
```

I have not tested thoroughly but it worked with simple examples.

Reply:
Nice that works! Changed to use external library.